general/react_dnd.md

```diff
@@ -0,0 +1,170 @@
+    </DndProvider>
+  )
+}
+```
+
+## useDrag
+
+First thing you need to configure is your draggable elements using useDrag hook.
+
+```
+const [{isDragging}, drag, preview] = useDrag({
+		collect: monitor => ({
+			isDragging: !!monitor.isDragging(),
+		}),
+		item: {
+            id, 
+            type: ACCEPTING_TYPES.ITEM
+        },
+	});
+```
+
+### Return value array
+
+`useDrag` returns an array of three values:
+
+- `Index 0`: An object containing collected properties from the collect function. If no collect function is defined, an empty object is returned.
+- `Index 1`: A connector function for the drag source. This **must** be attached to the draggable portion of the DOM.
+- `Index 2`: A connector function for the drag preview. This may be attached to the preview portion of the DOM in case you want to overwrite the default drag preview behavior.
+
+### Hook config object (most important properties)
+
+- `item`: Required. This is the only information about the draggable item that will be available for the drop or custom drag layer objects, so you need to place here any information you'll need on your drop logic.
+
+    `item.type` is also required, only drop targets registered for the same type will react to this drag.
+
+- `collect`: Optional. This is a function that receives `monitor` and `props`, and must return a plain object that you'll receive as the fist array element of the hook return value.
+
+### Connect drag with DOM element
+
+You can connect drag with an element in two differente ways:
+
+- Setting `drag` returned parameter as your element's `ref`.
+
+```
+return (
+    <div ref={drag}>
+        ...
+    </div>
+);
+```
+
+- Creating your own `ref`, calling `drag` returned parameter passing it out this ref and setting your element's ref with it.
+
+```
+const ref = useRef();
+
+drag(ref);
+
+return (
+    <div ref={ref}>
+        ...
+    </div>
+);
+```
+
+## useDrop
+
+Once the drag element is configured you can set your drop element(s) using `useDrop`.
+
+```
+const [{isOver}, drop] = useDrop({
+		accept: ACCEPTING_TYPES.ITEM,
+		canDrop(source, monitor) {
+			return isValidTarget(source, monitor);
+		},
+		collect: monitor => ({
+			isOver: !!monitor.isOver(),
+		}),
+		drop(source, monitor) {
+			if (monitor.canDrop()) {
+				onItemDrop(source, monitor);
+			}
+		},
+		hover(source, monitor) {
+			if (monitor.isOver() && monitor.canDrop()) {
+                onItemDropHover(source, monitor)
+			}
+		}
+	});
+```
+
+### Return value array
+
+`useDrop` returns an array of two values:
+
+- `Index 0`: An object containing collected properties from the collect function. If no collect function is defined, an empty object is returned.
+
+- `Index 1`: A connector function for the drop target. This **must** be attached to the drop-target portion of the DOM.
+
+### Hook config object (most important properties)
+
+- `accept`: Required. This must be set with the item type you want drop to react to.
```

Comment:
Hehe, I saw "react" here and immediately though you meant "React". Maybe "accept" would be better than "react to"?